src/utils/PaymentsConverter.ts

```diff
@@ -168,6 +162,71 @@ export function getPaymentsActivationResponse(
         datiPagamentoPA.enteBeneficiario.provinciaBeneficiario,
       nazioneBeneficiario: datiPagamentoPA.enteBeneficiario.nazioneBeneficiario
     },
-    spezzoniCausaleVersamento: datiPagamentoPA.spezzoniCausaleVersamento
+    spezzoniCausaleVersamento: getSpezzoniCausaleVersamentoForController(
+      datiPagamentoPA.spezzoniCausaleVersamento
+    )
   });
 }
+
+/**
+ * Define a IcodiceIdRPT object to send to PagoPA Services, containing payment information
+ * Ask the pagopa service administrator or read documentation from RptId definition
+ * @param {RptId} rptId - Payment information provided by BackendApp
+ * @return {PPTPortTypes.IcodiceIdRPT} The result generated for PagoPa
+ */
+function getCodiceIdRpt(rptId: RptId): PPTPortTypes.IcodiceIdRPT {
+  switch (rptId.paymentNoticeNumber.auxDigit) {
+    case "0":
+      return {
+        CF: rptId.organizationFiscalCode,
+        AuxDigit: rptId.paymentNoticeNumber.auxDigit,
+        CodStazPA: rptId.paymentNoticeNumber.applicationCode,
+        CodIUV: rptId.paymentNoticeNumber.iuv13
+      };
+    case "1":
+      return {
+        CF: rptId.organizationFiscalCode,
+        AuxDigit: rptId.paymentNoticeNumber.auxDigit,
+        CodIUV: rptId.paymentNoticeNumber.iuv17
+      };
+    case "2":
+      return {
+        CF: rptId.organizationFiscalCode,
+        AuxDigit: rptId.paymentNoticeNumber.auxDigit,
+        CodIUV: rptId.paymentNoticeNumber.iuv15
+      };
+    case "3":
+      return {
+        CF: rptId.organizationFiscalCode,
+        AuxDigit: rptId.paymentNoticeNumber.auxDigit,
+        CodIUV: rptId.paymentNoticeNumber.iuv13
+      };
```

Comment:
add a default case

Reply:
or use the `never` trick to check that the switch is exhaustive (ask @fgiobergia )

Reply:
If I remove a case, I get an error at compile time. Default case is not necessary because any cases should be covered to compile it.